File: perl/Sully.pl
#!/usr/bin/env perl

use strict;
use warnings;

use File::Basename;

my $FORMAT = '#!/usr/bin/env perl%1$c%1$cuse strict;%1$cuse warnings;%1$c%1$cuse File::Basename;%1$c%1$cmy $FORMAT = %3$c%4$s%3$c;%1$c%1$csub duplicate {%1$c%2$cmy ($i) = @_;%1$c%1$c%2$copen my $f, ">", "Sully_$i.pl" or die;%1$c%2$cprintf $f "$FORMAT", 10, 9, 39, "$FORMAT", $i;%1$c}%1$c%1$csub run {%1$c%2$cmy ($i) = @_;%1$c%1$c%2$csystem "/usr/bin/env perl Sully_$i.pl";%1$c}%1$c%1$csub main {%1$c%2$cmy $i = %5$d;%1$c%1$c%2$cif (basename(__FILE__) ne "Sully.pl") {%1$c%2$c%2$c$i -= 1;%1$c%2$c}%1$c%1$c%2$cduplicate $i;%1$c%1$c%2$cif ($i > 0) {%1$c%1$c%2$c%2$crun $i;%1$c%2$c}%1$c}%1$c%1$cmain();%1$c';

sub duplicate {
	my ($i) = @_;

	open my $f, ">", "Sully_$i.pl" or die;
	printf $f "$FORMAT", 10, 9, 39, "$FORMAT", $i;
}

sub run {
	my ($i) = @_;

	system "/usr/bin/env perl Sully_$i.pl";
}

sub main {
	my $i = 5;

	if (basename(__FILE__) ne "Sully.pl") {
		$i -= 1;
	}

	duplicate $i;

	if ($i > 0) {
		run $i;
	}
}

main();

File: python/Sully.py
#!/usr/bin/env python3

import os

FORMAT = "#!/usr/bin/env python3{1}{1}import os{1}{1}FORMAT = {0}{2}{0}{1}{1}def duplicate(i):{1}    with open(f{0}Sully_{{i}}.py{0}, {0}w{0}) as f:{1}        print(FORMAT.format(chr(34), chr(10), FORMAT, i), file=f){1}{1}def run(i):{1}    os.system(f{0}/usr/bin/env python3 Sully_{{i}}.py{0}) == 0 or exit(1){1}{1}if __name__ == {0}__main__{0}:{1}    i = {3}{1}{1}    if os.path.basename(__file__) != {0}Sully.py{0}:{1}        i -= 1{1}{1}    duplicate(i){1}    if i > 0:{1}        run(i)"

def duplicate(i):
    with open(f"Sully_{i}.py", "w") as f:
        print(FORMAT.format(chr(34), chr(10), FORMAT, i), file=f)

def run(i):
    os.system(f"/usr/bin/env python3 Sully_{i}.py") == 0 or exit(1)

if __name__ == "__main__":
    i = 5

    if os.path.basename(__file__) != "Sully.py":
        i -= 1

    duplicate(i)
    if i > 0:
        run(i)
